### PR TITLE
use CarActiveGuidanceMarkers in route preview screen

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description
While integrating the experimental route preview feature in Android Auto, I noticed that `CarRoutePreviewScreen` duplicates most of the logic in `CarActiveGuidanceMarkers`. That was resolved by using `CarActiveGuidanceMarkers` (should we rename it now?) in `CarRoutePreviewScreen`. Also working with styles using `handleStyleOnAttached`/`handleStyleOnDetached` was rather inconvenient, replaced that with `styleFlow`.

### CODE REVIEW

File: libnavui-androidauto/api/current.txt [hunk 1]
```diff
@@ -283,7 +283,7 @@ package com.mapbox.androidauto.map.logo {
 package com.mapbox.androidauto.navigation {
 
   public final class CarActiveGuidanceMarkers implements com.mapbox.maps.extension.androidauto.MapboxCarMapObserver {
-    ctor public CarActiveGuidanceMarkers();
+    ctor public CarActiveGuidanceMarkers(com.mapbox.androidauto.routes.CarRoutesProvider carRoutesProvider = com.mapbox.androidauto.routes.NavigationCarRoutesProvider());
```
Comment (kmadsen) [inline libnavui-androidauto/api/current.txt:286]: I thought we would be able to delete the `NavigationCarRoutesProvider` because `MapboxNavigation` will provide the routes

File: libnavui-dropin/api/current.txt [hunk 2]
```diff
@@ -245,6 +245,7 @@ package com.mapbox.navigation.dropin {
     method public com.mapbox.maps.plugin.annotation.generated.PointAnnotationOptions? getDestinationMarkerAnnotationOptions();
     method public Integer? getEndNavigationButtonStyle();
     method public Integer? getInfoPanelBackground();
+    method public Float? getInfoPanelGuidelineMaxPosPercent();
```
Comment (kmadsen) [inline libnavui-dropin/api/current.txt:248]: looks unrelated. there must be an issue with the api check

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/preview/CarRoutePreviewScreen.kt [hunk 3]
```diff
@@ -57,37 +49,7 @@ internal class CarRoutePreviewScreen @UiThread constructor(
         alternativeCarCameraMode = CarCameraMode.FOLLOWING,
         carRoutesProvider = carRoutesProvider,
     )
-
-    private var styleLoadedListener: OnStyleLoadedListener? = null
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/preview/CarRoutePreviewScreen.kt:61]: 💯

File: libnavui-androidauto/api/current.txt (hunk 1 above)
Comment (Zayankovsky) [reply libnavui-androidauto/api/current.txt:286]: I don't see how we can do that with the current route preview implementation in Nav SDK. But I can make the constructor with a parameter `internal` and only keep the parameter-less constructor `public`, what do you think?

File: libnavui-dropin/api/current.txt (hunk 2 above)
Comment (Zayankovsky) [reply libnavui-dropin/api/current.txt:248]: Someone forgot to update the files, when adding these APIs. But better late than never, right?

Comment (Zayankovsky) [reply libnavui-dropin/api/current.txt]: I rebased on `main`, this is no longer present in my PR.

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/extensions/MapboxCarMapSurfaceExtensions.kt [hunk 4]
```diff
@@ -5,30 +5,23 @@ package com.mapbox.androidauto.internal.extensions
 import com.mapbox.maps.Style
 import com.mapbox.maps.extension.androidauto.MapboxCarMapSurface
 import com.mapbox.maps.plugin.delegates.listeners.OnStyleLoadedListener
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
 
-fun MapboxCarMapSurface.handleStyleOnAttached(
-    block: (Style) -> Unit
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/extensions/MapboxCarMapSurfaceExtensions.kt:10]: 💯

File: libnavui-androidauto/api/current.txt (hunk 1 above)
Comment (kmadsen) [reply libnavui-androidauto/api/current.txt:286]: Yeah you're right, we also have the same thing to resolve with `CarRouteLineRenderer` and `CarNavigationCamera`

We are not entirely migrated yet to be able to remove this

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt [hunk 5]
```diff
@@ -1,63 +1,54 @@
...
-        val route = MapboxNavigationApp.current()?.getNavigationRoutes()?.firstOrNull()
-            ?: return
-        val coordinate = route.routeOptions.coordinatesList().lastOrNull()
-            ?: return
-        val featureCollection = FeatureCollection.fromFeature(Feature.fromGeometry(coordinate))
-        mapboxCarMapSurface?.getStyle()?.let {
-            placesLayerUtil.updatePlacesListOnMapLayer(it, featureCollection)
-        }
+    private fun updateMarkers(style: Style, routes: List<NavigationRoute>) {
+        val features = routes.take(1)
```
Comment (kmadsen) [inline libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt:30]: I like yours but also was thinking about alternatives

```
    private fun updateMarkers(style: Style, routes: List<NavigationRoute>) {
        val destination = routes.firstOrNull { it.routeOptions.coordinatesList().isNotEmpty() }
            ?.let { listOf(Feature.fromGeometry(it.routeOptions.coordinatesList().last())) }
            ?: emptyList<Feature>()
        placesLayerUtil.updatePlacesListOnMapLayer(style, FeatureCollection.fromFeatures(destination))
    }
```

**Review by kmadsen — APPROVED** [review]
Really nice clean up 🚀

File: libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt (hunk 5 above)
Comment (Zayankovsky) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt:30]: Yours is a bit different from what we had before.

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt:30]: is it? this is updating a destination marker right? I didn't test it i just considered logic the same

Comment (kmadsen) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt:30]: previously the null values would return. now it always calls `updatePlacesListOnMapLayer` so it seems there may have been an issue fixed 🤔

Comment (Zayankovsky) [reply libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/CarActiveGuidanceMarkers.kt:30]: You are right, mine is also different from what we had before. I think it is better to explicitly clear the markers in this case.
